### PR TITLE
Various bug fixes and improvements.

### PR DESCRIPTION
**Pull request by CodyFraker**
The #current-server-info embed is reaching its character limit. gotta figure out how to get the same info but less characters..

### CODE REVIEW

File: Bloon/Core/Commands/Attributes/GeneralAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                SBGChannels.General,
+                SBGChannels.Mapmaker,
+                SBGChannels.TradingPost,
+                SBGChannels.CompetitiveMatches,
+                SBGChannels.AUG,
+                SBGChannels.SecretBaseAlpha,
+                SBGChannels.Wiki,
+                SBGChannels.Offtopic,
+            };
+            if (availChannels.Contains(ctx.Channel.Id))
```
Comment (dukeofsussex) [inline Bloon/Core/Commands/Attributes/GeneralAttribute.cs:30]: Could be simplified to `return Task.FromResult(availChannels.Contains(ctx.Channel.Id)` (if you want to).

File: Bloon/Core/Commands/Attributes/StatsAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Bloon.Variables.Channels;
+    using DSharpPlus.CommandsNext;
+    using DSharpPlus.CommandsNext.Attributes;
+
+    /**
+     * <summary>
+     * Precondition for general commands.
```
Comment (dukeofsussex) [inline Bloon/Core/Commands/Attributes/StatsAttribute.cs:11]: Wrong.

File: Bloon/Core/Commands/Attributes/StatsAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            List<ulong> availChannels = new List<ulong>
+            {
+                SBGChannels.General,
+                SBGChannels.TradingPost,
+                SBGChannels.CompetitiveMatches,
+                SBGChannels.AUG,
+                SBGChannels.Wiki,
+                SBGChannels.Offtopic,
+            };
+            if (availChannels.Contains(ctx.Channel.Id))
```
Comment (dukeofsussex) [inline Bloon/Core/Commands/Attributes/StatsAttribute.cs:27]: Same as `GeneralAttribute`.

File: Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs [hunk 4]
```diff
@@ -12,6 +12,7 @@ namespace Bloon.Commands
     using DSharpPlus.CommandsNext.Attributes;
     using DSharpPlus.Entities;
 
+    [StatsAttribute]
```
Comment (dukeofsussex) [inline Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs:15]: Attribute could and should be moved to the feature.

File: Bloon/Features/Intruder Backend/Rooms/Commands/ServersCommand.cs [hunk 5]
```diff
@@ -9,6 +9,7 @@ namespace Bloon.Features.IntruderBackend.Servers
     using DSharpPlus.CommandsNext;
     using DSharpPlus.CommandsNext.Attributes;
     using DSharpPlus.Entities;
+    using Bloon.Utils;
```
Comment (dukeofsussex) [inline Bloon/Features/Intruder Backend/Rooms/Commands/ServersCommand.cs:12]: Remove pls.

File: Bloon/Features/User Events/WelcomeMatFeature.cs [hunk 6]
```diff
@@ -72,8 +72,45 @@ private async Task OnGuildMemberAddedAsync(DiscordClient dClient, GuildMemberAdd
             };
 
             await sbgChannel.SendMessageAsync(string.Empty, embed: userDetails);
-            sbgChannel = await this.dClient.GetChannelAsync(SBGChannels.General);
+        }
+
+        private async Task BloonsideEmbed(DiscordClient dClient, GuildMemberAddEventArgs args)
```
Comment (dukeofsussex) [inline Bloon/Features/User Events/WelcomeMatFeature.cs:77]: Copy-pasta that isn't called from the other method above.

File: Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs (hunk 4 above)
Comment (CodyFraker) [reply Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs:15]: attribute is used in more than one location tho for the features.

File: Bloon/Features/User Events/WelcomeMatFeature.cs (hunk 6 above)
Comment (CodyFraker) [reply Bloon/Features/User Events/WelcomeMatFeature.cs:77]: not supposed to be called from above. Forgot to add to the event.

Comment (dukeofsussex) [reply Bloon/Features/User Events/WelcomeMatFeature.cs:79]: Are there more stats commands that are going to use this?

File: Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs (hunk 4 above)
Comment (dukeofsussex) [reply Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs:15]: Are there more stats commands that are going to use this?

File: Bloon/Features/User Events/WelcomeMatFeature.cs (hunk 6 above)
Comment (dukeofsussex) [reply Bloon/Features/User Events/WelcomeMatFeature.cs:79]: Nice, still a copy-pasta though.

File: Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs (hunk 4 above)
Comment (CodyFraker) [reply Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs:15]: yes

File: Bloon/Features/User Events/WelcomeMatFeature.cs (hunk 6 above)
Comment (CodyFraker) [reply Bloon/Features/User Events/WelcomeMatFeature.cs:79]: i hand wrote this by short circuiting my usb 2.0 header.

File: Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs (hunk 4 above)
Comment (dukeofsussex) [reply Bloon/Features/Intruder Backend/Agents/Commands/Stats.cs:15]: Okily dokily

File: Bloon/Features/User Events/WelcomeMatFeature.cs (hunk 6 above)
Comment (dukeofsussex) [reply Bloon/Features/User Events/WelcomeMatFeature.cs:79]: Tell me lies, tell me sweet little lies.

**Review by dukeofsussex — APPROVED** [review]

**Pull request merged**